diff --git a/package/kernel/mt76/Makefile b/package/kernel/mt76/Makefile
index c0b276a..1397fae 100644
--- a/package/kernel/mt76/Makefile
+++ b/package/kernel/mt76/Makefile
@@ -6,10 +6,10 @@ PKG_RELEASE=1
 PKG_LICENSE:=GPLv2
 PKG_LICENSE_FILES:=
 
-PKG_SOURCE_URL:=https://github.com/openwrt/mt76
+PKG_SOURCE_URL:=https://github.com/Komodowifi/mt76.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2018-06-18
-PKG_SOURCE_VERSION:=f4506593349608e8a135ba7b465297260a7057f1
+PKG_SOURCE_DATE:=2018-08-06
+PKG_SOURCE_VERSION:=de3009b5b2ff80896363af3bfee2914d0cea31ab
 PKG_MIRROR_HASH:=00c8d564b6aed091728eb66f35d356c44cd90467f80666b4f19c9e71bb2fc290
 
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
@@ -50,6 +50,15 @@ define KernelPackage/mt76x2
   AUTOLOAD:=$(call AutoProbe,mt76x2e)
 endef
 
+define KernelPackage/mt76x0
+  $(KernelPackage/mt76-default)
+  TITLE:=MediaTek MT76x0 wireless driver
+  DEPENDS+=+kmod-mt76-core
+  FILES:=\
+    $(PKG_BUILD_DIR)/mt76x0e.ko
+#  AUTOLOAD:=$(call AutoProbe,mt76x0e)
+endef
+
 define KernelPackage/mt7603
   $(KernelPackage/mt76-default)
   TITLE:=MediaTek MT7603 wireless driver
@@ -92,6 +101,13 @@ define KernelPackage/mt76x2/install
 		$(1)/lib/firmware
 endef
 
+define KernelPackage/mt76x0/install
+	$(INSTALL_DIR) $(1)/lib/firmware
+	cp \
+		$(PKG_BUILD_DIR)/firmware/MT7610.bin \
+		$(1)/lib/firmware
+endef
+
 define KernelPackage/mt7603/install
 	$(INSTALL_DIR) $(1)/lib/firmware
 	cp $(if $(CONFIG_TARGET_ramips_mt76x8), \
@@ -105,6 +121,7 @@ define KernelPackage/mt7603/install
 endef
 
 $(eval $(call KernelPackage,mt76-core))
+$(eval $(call KernelPackage,mt76x0))
 $(eval $(call KernelPackage,mt76x2))
 $(eval $(call KernelPackage,mt7603))
 $(eval $(call KernelPackage,mt76))
